packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -264,34 +279,30 @@ function useFocusContainment(scopeRef: RefObject<HTMLElement[]>, contain: boolea
   }, [raf]);
 }
 
-function isElementInAnyScope(element: Element, scopes: Set<RefObject<HTMLElement[]>>) {
-  for (let scope of scopes.values()) {
+function isElementInAnyScope(element: Element) {
+  for (let scope of scopes.keys()) {
     if (isElementInScope(element, scope.current)) {
       return true;
     }
   }
   return false;
 }
 
-const focusScopeDataAttrNames = [
-  'data-focus-scope-start',
-  'data-focus-scope-end'
-];
-
-function isFocusScopeDirectChild(scope: HTMLElement) {
-  return focusScopeDataAttrNames.some(name => scope.getAttribute(name) !== null);
+function isElementInScope(element: Element, scope: HTMLElement[]) {
+  return scope.some(node => node.contains(element));
 }
 
-function isFocusScopeNestedChild(scope: HTMLElement) {
-  return focusScopeDataAttrNames.some(name => scope.querySelector(`[${name}]`));
-}
+function isAncestorScope(ancestor: ScopeRef, scope: ScopeRef) {
+  let parent = scopes.get(scope);
+  if (!parent) {
+    return false;
+  }
 
-function isFocusScopeInScope(scopes: HTMLElement[]) {
-  return scopes.some(scope => isFocusScopeDirectChild(scope) || isFocusScopeNestedChild(scope));
-}
+  if (parent === ancestor) {
+    return true;
+  }
 
-function isElementInScope(element: Element, scope: HTMLElement[]) {
-  return !isFocusScopeInScope(scope) && scope.some(node => node.contains(element));
+  return isAncestorScope(parent, scope);
```

Comment:
won't this just return true immediately? i think it's supposed to be this?
```suggestion
  return isAncestorScope(ancestor, parent);
```

Reply:
Yeah you're right, good catch.